File: src/main/java/cn/chahuyun/economy/model/bank/Bank.java
package cn.chahuyun.economy.model.bank;

import cn.chahuyun.economy.model.bank.action.BankAction;

/**
 * @author Erzbir
 * @Date: 2022/11/29 21:55
 */
public class Bank extends AbstractBank {
    public static final Bank INSTANCE = new Bank();

    private Bank() {
        super();
    }

    public void execute(BankAction action) throws Exception {
        action.execute();
    }

}

File: economy-core/src/main/kotlin/cn/chahuyun/economy/model/bank/Bank.kt
package cn.chahuyun.economy.model.bank

import cn.chahuyun.economy.model.bank.action.BankAction

/**
 * @author Erzbir
 * @Date: 2022/11/29 21:55
 */
class Bank private constructor() : AbstractBank() {
    companion object {
        @JvmField
        val INSTANCE = Bank()
    }

    @Throws(Exception::class)
    fun execute(action: BankAction) {
        action.execute()
    }
}

